samples_project/Assets/SampleViewer/Samples/WeatherQuery/Scripts/WeatherQuery.cs

```diff
@@ -0,0 +1,250 @@
+            }
+            else if (feature.properties.WEATHER.ToLower().Contains("snow"))
+            {
+                scrollItem.weatherImage.sprite = Snow;
+            }
+            else if (feature.properties.WEATHER.ToLower().Contains("rain"))
+            {
+                scrollItem.weatherImage.sprite = Rain;
+            }
+            else if (feature.properties.SKY_CONDTN.ToLower().Contains("cloud"))
+            {
+                scrollItem.weatherImage.sprite = Cloudy;
+            }
+            else
+            {
+                scrollItem.weatherImage.sprite = Sunny;
+            }
+        }
+    }
+
+    public void SelectItems(ScrollItem ScrollItem)
+    {
+        foreach (var toggle in ListItems)
+        {
+            toggle.isOn = false;
+        }
+
+        ScrollItem.isEnabled = true;
+    }
+
+    public IEnumerator SendCityQuery(double x, double y)
+    {
+        string APIToken;
+        string query;
+        ArcGISMapComponent mapComponent = GameObject.FindObjectOfType<ArcGISMapComponent>();
+
+        if (mapComponent.APIKey != "")
+        {
+            APIToken = mapComponent.APIKey;
+        }
+        else
+        {
+            APIToken = ArcGISProjectSettingsAsset.Instance.APIKey;
+        }
+
+        string url = "https://geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer/reverseGeocode";
+        string coordinates = x.ToString() + "," + y.ToString();
+        query = url + "/?f=json&token=" + APIToken + "&location=" + coordinates;
+        UnityWebRequest Request = UnityWebRequest.Get(query);
+        yield return Request.SendWebRequest();
+
+        if (Request.result != UnityWebRequest.Result.Success)
+        {
+            Debug.Log(Request.error);
+        }
+        else
+        {
+            var deserialized = JsonUtility.FromJson<CityData>(Request.downloadHandler.text);
+
+            if (deserialized.address.City != "")
+            { 
+                cityText.text = deserialized.address.City + ", " + deserialized.address.Region; 
+                notFound = false;
+            }
+            else
+            {
+                notFound = true;
+            }
+        }
+    }
+
+    public void SetTempType()
+    {
+        if (TempuratureToggle.isOn)
+        {
+            tempTypeText.text = "C";
+        }
+        else
+        {
+            tempTypeText.text = "F";
+        }
+    }
+
+    private void Start()
+    {
+        StartCoroutine(GetFeatures());
+
+        dropDownButton.onValueChanged.AddListener(delegate (bool value)
+        {
+            outfieldsList.SetActive(value);
+            if (!value)
+            {
+                dropDownAnim.Play("ReverseDropDownArrow");
+            }
+            else
+            {
+                dropDownAnim.Play("DropDownArrow");
+            }
+        });
+
+    }
+
+    private void Update()
+    {
+        arcGISCamera.gameObject.GetComponent<ArcGISCameraControllerComponent>().enabled = !MouseOverUI();
```

Comment:
Is there a decent way to do this with an event?

Reply:
Not without creating a new script that gets attached to the UI to check on enter. This is how we do it in all the samples the require this, which is most of them.